lib/jxl/enc_modular.cc

```diff
@@ -321,6 +320,106 @@ bool maybe_do_transform(Image& image, const Transform& tr,
   return did_it;
 }
 
+void try_palettes(Image& gi, int& max_bitdepth, int& maxval,
+                  const CompressParams& cparams_, float channel_colors_percent,
+                  jxl::ThreadPool* pool = nullptr) {
+  float cost_before = 0.f;
+  size_t did_palette = 0;
+  float nb_pixels = gi.channel[0].w * gi.channel[0].h;
+
+  if (cparams_.palette_colors != 0 || cparams_.lossy_palette) {
+    // when not estimating, assume 5 bpp (arbitrarily)
```

Comment:
Shouldn't that 5 depend on bitdepth and channel count?
Say, 0.1 * bitdepth * nb_chan would seem like a reasonable estimate...

Reply:
Makes sense. Made it twice that, so it's 4.8 for 8-bit RGB. It doesn't need to be a good estimate, it only affects the choice of max_colors anyway.